src/discord/commands/moderation/nickname.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    Command, AccessLevel,
+    CommandError, Logger,
+    Guards
+} from "dd-botkit";
+
+export const NicknameCommand: Command = {
+    opts: {
+        name: "nickname",
+        access: AccessLevel.MODERATOR,
+        category: "Moderation",
+        guards: [
+            Guards.Argumented(
+                "nickname",
+                "Force changes the nickname of a user",
+                [{
+                    name: "user",
+                    type: "user",
+                    required: true
+                },
+                {
+                    name: "nickname",
+                    type: "string",
+                    required: true
+                }]
+            )
+        ]
+    },
+    handler: async (msg, next) => {
+        const [user, nickname] = msg.args;
+
+          await msg.guild.members.get((user as any).id).setNickname(nickname)
```

Comment:
If the `get` call returns undefined or null, you will get a runtime error (i.e if the ID doesn't exist). That should be handled.

Reply:
Your options are to either ensure the user is in that guild or check whether it is undefined or null. Moreover, please do `user as User` instead of `user as any`. BotKit `1.0.10` has the proper typings.

Reply:
@olvrb @nullpixel told me that it was ok to not do any error catching because it would auto catch. It would be impossible to fail as well due to `user` being in a required state.

Reply:
These are not the types of errors that should be left uncaught. Errors are auto-caught to allow people to throw [CommandError](https://github.com/dynastic/BotKit/blob/master/src/commands/errors.ts) which is nicely transformed into a presentable error message for users.

Reply:
Resolved, updated commit to new code.